test/tools/libtesteth/BlockChainHelper.h

```diff
@@ -137,10 +137,19 @@ class TestBlock
 class TestBlockChain
 {
 public:
-    TestBlockChain() : TestBlockChain(defaultGenesisBlock()) {}
-    TestBlockChain(TestBlock const& _genesisBlock);
+    enum MiningType
+    {
+        Ethash,
+        NoProof
+    };
+    TestBlockChain() : TestBlockChain(defaultGenesisBlock(), MiningType::Ethash) {}
+    TestBlockChain(TestBlock const& _genesisBlock, MiningType _mining = MiningType::Ethash);
+
+    // Replace sealEngine depending on a MiningType"
+    // Genesis config is parsed from _net in libethashseal/GenesisInfo.cpp
+    static std::string prepareGenesisConfig(eth::Network _net, MiningType _mining);
```

Comment:
it is used in ImportTest.cpp when generating the bc out of state test.
however if it is possible to change sealengine in already parese chain params it might not be needed at all

Reply:
Yeah I guess you should be able to modify `ChainOperationParams::sealEngineName`